src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt

```diff
@@ -192,8 +194,15 @@ internal class RiskService(
         historical = null,
       )
     }
-    val latestScores = riskScoresResponse.sortedBy { LocalDateTime.parse(it.completedDate) }.reversed().firstOrNull()
-    val historicalScores = riskScoresResponse.sortedBy { it.completedDate }.reversed()
+    val latestScores = riskScoresResponse
+      .filter { !it.completedDate.isNullOrEmpty() }
+      .sortedBy { LocalDateTime.parse(it.completedDate) }
+      .reversed()
+      .firstOrNull()
+    val historicalScores = riskScoresResponse
+      .filter { !it.completedDate.isNullOrEmpty() }
+      .sortedBy { LocalDateTime.parse(it.completedDate) }
+      .reversed()
```

Comment:
This is doing the same operation twice, duplicating code and processing, and risking getting a different result (either because the code isn't updated correctly or because of non-determinism in the result).  It should be sorted once, then from that list, take the first.

Reply:
Corrected 😌

Reply:
We don't need the map